### PR TITLE
chore(provider): add more test coverage on `BlockchainProvider::*by_block_range` queries

### PR DESCRIPTION
**Pull request by joshieDo**
follow up to https://github.com/paradigmxyz/reth/pull/11480

follows the same logic from the PR above but applies it to block ranges

cc @tcoratger deleted a bunch of tests as a consequence of `test_methods_by_block_range`. lmk if you think there's some coverage which would get lost this way

### CODE REVIEW

**Review by mattsse — APPROVED** [review]
lgtm!

**Comment by tcoratger** [comment]
@joshieDo Let me check this quickly

File: crates/storage/provider/src/providers/blockchain_provider.rs [hunk 1]
```diff
@@ -4207,6 +3744,103 @@ mod tests {
...
+
+                // Test range that spans database and in-memory with unbounded end
+                // assert_eq!(
+                //     $provider.$method(in_mem_range.start() - 2..)?,
+                //     database_data[database_data.len() - 2..]
+                //         .iter()
+                //         .chain(&in_memory_data[..])
+                //         .cloned()
+                //         .collect::<Vec<_>>()
+                // );
```
Comment (tcoratger) [inline crates/storage/provider/src/providers/blockchain_provider.rs:3793]: Do we need to uncomment this or keep as is?

File: crates/storage/provider/src/providers/blockchain_provider.rs [hunk 2]
```diff
@@ -4207,6 +3744,103 @@ mod tests {
...
+
+                // Test empty range
+                let start_block_num = u64::MAX;
+                let end_block_num = u64::MAX;
+                let result = $provider.$method(start_block_num..=end_block_num-1)?;
+                assert!(result.is_empty(), "No data should be found for an empty block range");
```
Comment (tcoratger) [inline crates/storage/provider/src/providers/blockchain_provider.rs:3799]: Maybe here you don't want to test this behavior but I thought it might be even better to add below a test with a non-empty range but that does not correspond to anything valid. For example with `start_block_num = 1000` and `end_block_num = 5000` so as to check that if the range is not empty but does not contain a block then the result must also be empty.

Even if the behavior is close I say that to validate the behavior that the range is empty or that it is not empty but with no block in it (the behaviors of the functions can potentially be different in both cases).

**Review by tcoratger — APPROVED** [review]
